### DISCUSSION TITLE
Vercel Deployment issues

### DISCUSSION THREAD

**Discussion by sontiO** [prisma/prisma#12849] [Q&A]
Hey everyone, struggling a bit with deploying my prisma project. In development everything is working for me but once deployed, I face the following error once navigating to a site that uses prisma data:
```

500
ENOENT: no such file or directory, open '/var/task/schema.prisma'
Error: ENOENT: no such file or directory, open '/var/task/schema.prisma'
```

In the vercel logs I can see that the users I fetch from prisma are undefined.

heres my code

`//schema.prisma:
datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}`

`//.env:
DATABASE_URL=mongodb+srv://admin:password@cluster0.kltzu.mongodb.net/users_test?retryWrites=true&w=majority
`
So far so good for development. 
According to prisma and vercel docs, my next steps were the following. 

In vercel dashoard UI, I set my enviroment variable DATABASE_URL like this  
"mongodb+srv://admin:password@cluster0.kltzu.mongodb.net/users_test". It says my env will be available as process.env.API_URL, but where do I use this? I think I am missing a step.

**Comment by austincrim** [comment]
Hey @sontiO 👋 ,

Could you share the structure of your repo?

> It says my env will be available as process.env.API_URL

You should be good as long as you have the `DATABASE_URL` set in the Vercel dashboard and have your `schema.prisma` datasource set up for `env(DATABASE_URL)`.

**Comment by janpio** [comment]
The `.env` file is usually not committed to your git repository, so would not be available on a deployment on Vercel. There you set the value of `DATABASE_URL` via their secrets UI and then when you app is deployed they load that and make it available to your app.

The error message you are getting is very different though and does not indicate a problem your environment variables. How are you deploying your app? How are you building your Prisma Client?

**Reply by vladi-stevanovic** [reply]
👋 Hello @sontiO  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by rishabhthakur11** [comment]
Hey @vladi-stevanovic i am facing the similar issue as @sontiO is facing

**Reply by gloriliale** [reply]
@rishabhthakur11 Are you running into this issue while using `getSession` in an API route? I found that the suggestion of using `getServerSession` in this thread solved it:  https://github.com/prisma/prisma/discussions/19400

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
